pylabnet/network/client_server/toptica_dl_pro.py

```diff
@@ -6,110 +6,110 @@
 
-    def exposed_temp_sp(self):
+    def exposed_temp_sp(self, laser_num):
         """ Gets temperature setpoint
 
         :return: (float) value of temperature setpoint
         """
 
-        return self._module.temp_sp()
+        return self._module.temp_sp(laser_num)
 
-    def exposed_temp_act(self):
+    def exposed_temp_act(self, laser_num):
         """ Gets actual DL temp
 
         :return: (float) value of temperature
         """
 
-        return self._module.temp_act()
+        return self._module.temp_act(laser_num)
 
-    def exposed_set_temp(self, temp):
+    def exposed_set_temp(self, temp, laser_num):
         """ Sets the current to desired value
-        
+
         :param temp: (float) value of temperature to set to in Celsius
         """
 
         temp = pickle.loads(temp)
-        return self._module.set_temp(temp)
+        return self._module.set_temp(temp, laser_num)
 
-    def exposed_configure_scan(self, offset=65, amplitude=100, frequency=0.2):
+    def exposed_configure_scan(self, offset=65, amplitude=100, frequency=0.2, laser_num=1):
         """ Sets the scan parameters for piezo scanning
 
         :param offset: (float) scan offset (center value) in volts (between 0 and 130)
         :param amplitude: (float) scan amplitude (peak to peak) in volts
         """
 
-        return self._module.configure_scan(offset, amplitude, frequency)
+        return self._module.configure_scan(offset, amplitude, frequency, laser_num)
 
-    def exposed_start_scan(self):
+    def exposed_start_scan(self, laser_num=1):
         """ Starts a piezo scan """
 
-        return self._module.start_scan()
+        return self._module.start_scan(laser_num)
 
-    def exposed_stop_scan(self):
+    def exposed_stop_scan(self, laser_num=1):
         """ Stops piezo scan """
 
-        return self._module.stop_scan()
+        return self._module.stop_scan(laser_num)
 
 
 
 class Client(ClientBase):
 
-    def is_laser_on(self):
-        return self._service.exposed_is_laser_on()
+    def is_laser_on(self, laser_num=1):
+        return self._service.exposed_is_laser_on(laser_num)
 
-    def turn_on(self):
-        return self._service.exposed_turn_on()
+    def turn_on(self, laser_num=1):
+        return self._service.exposed_turn_on(laser_num)
 
-    def turn_off(self):
-        return self._service.exposed_turn_off()
+    def turn_off(self, laser_num=1):
+        return self._service.exposed_turn_off(laser_num)
 
-    def voltage(self):
-        return self._service.exposed_voltage()
+    def voltage(self, laser_num=1):
+        return self._service.exposed_voltage(laser_num)
 
-    def set_voltage(self, voltage):
+    def set_voltage(self, voltage, laser_num=1):
         voltage = pickle.dumps(voltage)
-        return self._service.exposed_set_voltage(voltage)
+        return self._service.exposed_set_voltage(voltage, laser_num)
 
     def set_ao_voltage(self, ao_channel=[], voltages=[0]):
```

Comment:
this method is used to fill the wrapper for generic AO hardware use in the wavemeter software. It should be adapted to be compatible by changing to:
```python
def set_ao_voltage(self, ao_channel=1, voltages=[0]):
    voltage = pickle.dumps(voltages[0])
    return self._service.exposed_set_voltage(voltage, ao_channel)
```